qtcodes/fitters/rotated_surface.py

```diff
@@ -23,6 +24,23 @@ class RotatedDecoder(LatticeDecoder):
     encoder_type = XXZZQubit
     syndrome_graph_keys = ["X", "Z"]
 
+    def _params_validation(self):
+        super()._params_validation()
+
+        # validation
+        if isinstance(self.params["d"], float) or isinstance(self.params["d"], int):
+            self.params["d"] = (int(self.params["d"]), int(self.params["d"]))
```

Comment:
I have similar code in mine if you want to crib:
```
        if isinstance(self.params["d"], Number):
            d = self.params["d"]
            self.params["d"] = (d, d)
```

But I'm also curious about the casts. Are you anticipating incorrect types being passed in? Here's something general purpose for the above though:
```
        d = tuple(map(int, self.params["d"]))
```

Reply:
So, I'm doing a typecast to `int` because many areas of the code expect `d` to be an integer (e.g. indexing or ranging indices in a for loop, etc.). Moreover, `d` cannot be a float, on principle.